<launch>

    <arg name="scan_topic"  default="/armbot_laser_scanner/laser/scan" />
    <arg name="base_frame"  default="base_link"/>
    <arg name="odom_frame"  default="odom"/>

    <!-- Run Gmapping --> 
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
        <remap from="scan" to="$(arg scan_topic)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="odom_frame" value="$(arg odom_frame)"/>
        
        <param name="map_update_interval" value="2.0"/>        
        <param name="maxUrange" value="50.0"/>
        <param name="maxRange" value="50.0"/>
        <param name="sigma" value="0.05"/>
        <param name="kernelSize" value="1"/>
        <param name="lstep" value="0.05"/>
        <param name="astep" value="0.05"/>
        <param name="iterations" value="5"/>
        <param name="lsigma" value="0.075"/>
        <param name="ogain" value="3.0"/>
        <param name="lskip" value="0"/>
        <param name="srr" value="0.01"/>
        <param name="srt" value="0.02"/>
        <param name="str" value="0.01"/>
        <param name="stt" value="0.02"/>
        <param name="linearUpdate" value="0.01"/>
        <param name="angularUpdate" value="0.01"/>
        <param name="temporalUpdate" value="0.1"/>
        <param name="resampleThreshold" value="0.5"/>
        <param name="particles" value="30"/>
        <param name="xmin" value="-5.0"/>
        <param name="ymin" value="-5.0"/>
        <param name="xmax" value="5.0"/>
        <param name="ymax" value="5.0"/>
        <param name="delta" value="0.1"/>
        <param name="llsamplerange" value="0.01"/>
        <param name="llsamplestep" value="0.01"/>
        <param name="lasamplerange" value="0.005"/>
        <param name="lasamplestep" value="0.005"/>
        <param name="minimumScore" value="0.005"/>        
    </node>

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <!-- <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/> -->
        <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/>        
        <rosparam file="$(find armbot_nav)/config/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
        <rosparam file="$(find armbot_nav)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find armbot_nav)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find armbot_nav)/config/global_costmap_params.yaml" command="load" /> 
        <rosparam file="$(find armbot_nav)/config/base_local_planner_params.yaml" command="load" />
        <remap from="/cmd_vel" to="/armbot/mobile_base_controller/cmd_vel"/>
        <remap from="/odom" to="/armbot/mobile_base_controller/odom"/>
    </node>

</launch> 